### ISSUE TITLE
Embedded geostory URL issue with scroll position

### ISSUE THREAD

**Issue by ElenaGallo** [geosolutions-it/MapStore2#9047]
## Description

The embedded geostory is not visible if the _Advanced Options_ of the direct share link are enable.

## How to reproduce

-  Open a geostory and follow the steps in the video below:
 

https://user-images.githubusercontent.com/56537133/226941975-fa26208f-9cd9-48a5-9cc3-5323b990b23a.mov

*Expected Result*

The URL of the embedded geostory does not contain the scroll position

**Comment by dsuren1** [comment]
**Investigation**
This is **_not_** a regression (also happens in **v2022.02.00**), as there was no implementation that specifically handles the removal of the query params when url generated for `geostory-embedded`. By default all the query params applicable for shared url is present on embedded url.

We can add a specific config in `embedOptions` perhaps to remove query params from shared url to form a embedded-url, there by handling this scenario in an isolated manner
localConfig: 
```
{
  geostory: [
    {
      "name": "Share",
      "cfg": {
        "embedOptions": {
          "removeQueryFromShared": true
        }
      }
    }
  ]
}
```
@tdipisa Kindly let me know your thoughts.

**Comment by tdipisa** [comment]
@dsuren1 thank you.

>as there was no implementation that specifically handles the removal of the query params when url generated for geostory-embedded. By default all the query params applicable for shared url is present on embedded url.

I'm sorry but it is not clear to me. What do you mean precisely? The expected behavior in this case would not be to 'remove' the section path in case of embedded URL. The expected behavior would be to have it working by scrolling to the desired section also for embedded gestories.

**Comment by ElenaGallo** [comment]
@dsuren1 it's my mistake, sorry. @tdipisa  is right. The expected behavior would be to have it working by scrolling to the desired section also for embedded geostories.

**Comment by tdipisa** [comment]
> As per the issue description, the expected result is
> 
> ```
> The URL of the embedded geostory does not contain the scroll position and the home button
> ```

@dsuren1 
you are right, I'm sorry. I did't see it and probably I forgot that there is some missing implementation that is preventing the embedded URL to work with _/section/id_  endpoint. I don't remember why precisely, we should check, but it is not the case here now since for this issue the identified behavior is effectively a regression from what we previously released. 
I can confirm that because I've tested with binaries of previous versions until 2021.01.03 where effectively the _/section/id_ was not included in the embedded URL. The **showHome** query parmeter yes instead, but it is not preventing the embedded URL to work as expected: that parameter  is simply ignored if present.

**Comment by dsuren1** [comment]
@tdipisa 

![section-id](https://user-images.githubusercontent.com/26929983/227192508-932b5b5e-0022-4b67-bf1c-777ad168ec67.gif)

From the gif, you can see that section id is not added when user is on the first section but added when scrolled to second. This is because the section id and column id is parsed from the url [here ](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/utils/ShareUtils.js#L113) instead of the fetching the value from `state->geostory-> currentPage`
![image](https://user-images.githubusercontent.com/26929983/227192440-bbf1efbe-f673-4701-9a7e-2ce16575bfac.png)

There are two issues,

- The section id is not added to shared url when user is on title/first page of geostory i.e when url doesn;t have section/column info
- The section id is not updated in the url when user scrolls with `Include scroll position` option enabled

As I mentioned above, there is no implementation to remove section id from geostory-embedded url when `Include scroll position` is enabled that adds section id to sharedUrl
